test/builder/eventlistener.go

```diff
@@ -60,19 +64,47 @@ func EventListenerServiceAccount(saName string) EventListenerSpecOp {
 	}
 }
 
-// EventListenerTrigger adds a Trigger to the EventListenerSpec Triggers.
-func EventListenerTrigger(tbName, ttName, apiVersion string) EventListenerSpecOp {
+// EventListenerTrigger adds an EventListenerTrigger to the EventListenerSpec Triggers.
+// Any number of EventListenerTriggerOp modifiers can be passed to create/modify it.
+func EventListenerTrigger(tbName, ttName, apiVersion string, ops ...EventListenerTriggerOp) EventListenerSpecOp {
 	return func(spec *v1alpha1.EventListenerSpec) {
-		spec.Triggers = append(spec.Triggers,
-			v1alpha1.Trigger{
-				TriggerBinding: v1alpha1.TriggerBindingRef{
-					Name:       tbName,
-					APIVersion: apiVersion,
-				},
-				TriggerTemplate: v1alpha1.TriggerTemplateRef{
-					Name:       ttName,
-					APIVersion: apiVersion,
+		spec.Triggers = append(spec.Triggers, Trigger(tbName, ttName, apiVersion, ops...))
+	}
+}
+
+// EventListenerTriggerParam adds a param to the EventListenerTrigger
+func EventListenerTriggerParam(name, value string) EventListenerTriggerOp {
+	return func(trigger *v1alpha1.EventListenerTrigger) {
+		trigger.Binding.Params = append(trigger.Binding.Params,
+			pipelinev1.Param{
+				Name: name,
+				Value: pipelinev1.ArrayOrString{
+					StringVal: value,
+					Type:      pipelinev1.ParamTypeString,
 				},
 			})
 	}
 }
+
+// Trigger creates an EventListenerTrigger. Any number of EventListenerTriggerOp
+// modifiers can be passed to create/modify it.
+func Trigger(tbName, ttName, apiVersion string, ops ...EventListenerTriggerOp) v1alpha1.EventListenerTrigger {
```

Comment:
Before the trigger was created within the `EventListenerTrigger()` function. This seems slightly confusing to me according to that function's docstring. Could you do the below:
```go
 		for _, op := range ops {
		op(&t)
	}
```
within that function?

Reply:
The `EventListenerTrigger()` function returned (and still does return) an `EventListenerSpecOp`. However, I wanted to use the builder functions to create an `EventListenerTrigger` in some of the lower-level unit tests that don't use an `EventListener` directly. So, I created the `Trigger()` function which is used as a helper in `EventListenerTrigger()`, and can also create `EventListenerTrigger` objects in the unit tests.